### ISSUE TITLE
Replace HarmonicAngleForce with AngleForce (and likewise for bonds?) in SMIRFF

### ISSUE THREAD

**Issue by davidlmobley** [open-forcefield-group/smarty#179]
We need to replace the `HarmonicAngleForce` SMIRFF tag with `AngleForce`, which would default to harmonic unless otherwise specified ( to allow the format to support other forms if desired), as per @jchodera discussion with Swope and others at IBM.

@jchodera - presumably we should do the same for bonds as well, to potentially allow support for Morse potentials, etc.?

**Comment by jchodera** [comment]
Yes, I think we should generalize bonds and angles this way.

We should also generalize `NonbondedForce` to permit other forms (e.g. Buckingham exp-6, Halgren buffered 14-7, etc.).

**Comment by davidlmobley** [comment]
> We should also generalize NonbondedForce to permit other forms (e.g. Buckingham exp-6, Halgren buffered 14-7, etc.).

Yes. I didn't mention that here, as `HarmonicAngleForce` and `HarmonicBondForce` are tags in the format which need changing, whereas the `NonbondedForce` tag doesn't need changing (I don't think) we just need to change how it's implemented. In other words, changing bonds/angles requires modifying our current code whereas handling other nonbonded forms just requires generalizing/extending. So I figured that was a separate issue.

(I'm working up a PR to the format spec which would address these issues.)

**Comment by jchodera** [comment]
Could address #42 (versioning) as well!

**Comment by davidlmobley** [comment]
On versioning - I commented on #42.

**Comment by jchodera** [comment]
Addressed in https://github.com/openforcefield/openforcefield/pull/86
